src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+const getVisibleTodos = (
+  todos: Todo[],
+  sortType: string,
+  query: string,
+): Todo[] => {
+  let filtered = todos;
+
+  if (sortType === 'active') {
+    filtered = todos.filter(todo => !todo.completed);
+  }
+
+  if (sortType === 'completed') {
+    filtered = todos.filter(todo => todo.completed);
+  }
+
+  return filtered.filter(todo => (
+    todo.title.toLowerCase().includes(query.toLowerCase())
+  ));
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeTodo, setActiveTodo] = useState<Todo>();
+  const [sortType, setSortType] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => setTodos(todosFromServer));
+  }, []);
+
+  const changeActiveTodo = (id: number) => {
+    setActiveTodo(todos.find(todo => todo.id === id));
+  };
```

Comment:
Can be optimized with `useCallback`, because you're passing this as prop to every todo in the `TodoList`